### PR TITLE
Build CLI

### PR DESCRIPTION
**Pull request by aylei**
This is a very early version of our CLI tool, for better reviewing experience, debug related functionalities are moved to next PR.

Some decisions are vague and supposed to be changed later on, namely, the CLI tool is currently alpha, however, I think the basic code skeleton is almost fine.

Moreover, this PR updates kubernetes related dependencies to 1.12.5

PTAL @tennix @xiaojingchen @weekface @shuijing198799 @gregwebs 

- [x] Update kubernetes dependencies to 1.12.5
- [x] tidb context sync and switch
- [x] tidb cluster view
- [x] volumes management
- [x] tidb cluster troubleshooting (moved to next PR for better reviewing)

Some examples in my local machine:
```shell
$ tkc
"tkc"(TiDB kubernetes control) is a command line interface for cloud tidb management and troubleshooting.

Cluster Management Commands:
  list        list all tidb clusters
  get         get all tidb clusters
  info        Show tidb cluster information.
  use         Specify a tidb cluster to use

Troubleshooting Commands:
  debug       Not implemented
  pdctl       Not implemented
  ctop        Not implemented

Other Commands:

Usage:
  tkc [flags] [options]

Use "tkc <command> --help" for more information about a given command.
Use "tkc options" for a list of global command-line options (applies to all commands).

$ tkc list
NAMESPACE      NAME                PD    TIKV     TIDB     AGE
dengshuan      demo                3/3   3/3      2/2      20d

$ tkc list -A
NAMESPACE      NAME                PD    TIKV     TIDB     AGE
chenqingjing   a                   1/1   3/3      1/1      4d5h
chenshuning    syncer-test         3/3   5/5      2/2      75d
dengshuan      demo                3/3   3/3      2/2      20d
developer      stn-qps             0/1   3/3      0/0      193d
liuzhi         linuxgit            2/3   3/3      2/2      157d
qupeng         samsung-stability   3/3   1/4      1/2      134d
wangke         newwktidb           3/3   1/3      2/3      39d
wangke         tidbwk              0/3   <none>   <none>   46d
wentaojin      marvin              3/3   3/3      2/2      27d
wuyelei        aylei-master        3/3   5/5      4/4      29d
xianjingwang   demo                0/1   <none>   <none>   39d
yueyuezhou     demo                0/3   <none>   <none>   27d
zhangbokang    zbk                 1/1   4/4      1/2      154d
zhengrong      money2              0/1   0/4      5/6      39d

$ tkc use -n=wuyelei aylei-master
Tidb cluster switched to wuyelei/aylei-master

$ tkc list
NAME           PD    TIKV   TIDB   AGE
aylei-master   3/3   5/5    4/4    29d

$ tkc info
Name:               aylei-master
Namespace:          wuyelei
CreationTimestamp:  2019-03-04 14:43:52 +0800 CST
Overview:
         Phase   Ready  Desired  CPU  Memory  Storage  Version
         -----   -----  -------  ---  ------  -------  -------
  PD:    Normal  3      3                     1Gi      pingcap/pd:v2.1.0
  TiKV:  Normal  5      5                     10Gi     pingcap/tikv:v2.1.0
  TiDB   Normal  4      4                              pingcap/tidb:v2.1.0
Endpoints(NodePort):
  - 172.16.30.10:37627
  - 172.16.30.7:37627
  - 172.16.30.9:37627
  - 172.16.10.11:37627

$ tkc get tikv
tkc get tikv
NAME                  READY   STATUS    MEMORY       CPU   RESTARTS   AGE   NODE
aylei-master-tikv-0   2/2     Running   50Mi/100Mi         0          29d   172.16.10.19
aylei-master-tikv-1   2/2     Running   50Mi/100Mi         0          29d   172.16.30.30
aylei-master-tikv-2   2/2     Running   50Mi/100Mi         3          29d   172.16.10.12
aylei-master-tikv-3   2/2     Running   50Mi/100Mi         0          21d   172.16.10.5
aylei-master-tikv-4   2/2     Running   50Mi/100Mi         0          28d   172.16.10.11

$ tkc get volume
VOLUME              CLAIM                      STATUS   CAPACITY   NODE           LOCAL
local-pv-1896e6ce   pd-aylei-master-pd-2       Bound    278Gi      172.16.10.14   /mnt/sas-disks/vol2
local-pv-2edc6512   pd-aylei-master-pd-1       Bound    1861Gi     172.16.30.28   /mnt/sas-disks/vol1
local-pv-482a865d   tikv-aylei-master-tikv-3   Bound    278Gi      172.16.10.5    /mnt/sas-disks/vol1
local-pv-5bbb9350   tikv-aylei-master-tikv-2   Bound    278Gi      172.16.10.12   /mnt/sas-disks/vol2
local-pv-5e34b56e   tikv-aylei-master-tikv-0   Bound    278Gi      172.16.10.19   /mnt/sas-disks/vol3
local-pv-6a1229d6   tikv-aylei-master-tikv-4   Bound    278Gi      172.16.10.11   /mnt/sas-disks/vol3
local-pv-84f137b4   pd-aylei-master-pd-0       Bound    1861Gi     172.16.30.29   /mnt/sas-disks/vol2
local-pv-fda68cc8   tikv-aylei-master-tikv-1   Bound    1861Gi     172.16.30.30   /mnt/sas-disks/vol0

$ tkc get all
NAME                READY   STATUS    MEMORY          CPU   RESTARTS   AGE   NODE
aylei-master-pd-0   1/1     Running   <none>/<none>         0          29d   172.16.30.29
aylei-master-pd-1   1/1     Running   <none>/<none>         2          29d   172.16.30.28
aylei-master-pd-2   1/1     Running   <none>/<none>         1          29d   172.16.10.14
aylei-master-tikv-0   2/2   Running   50Mi/100Mi         0     29d   172.16.10.19
aylei-master-tikv-1   2/2   Running   50Mi/100Mi         0     29d   172.16.30.30
aylei-master-tikv-2   2/2   Running   50Mi/100Mi         3     29d   172.16.10.12
aylei-master-tikv-3   2/2   Running   50Mi/100Mi         0     21d   172.16.10.5
aylei-master-tikv-4   2/2   Running   50Mi/100Mi         0     28d   172.16.10.11
aylei-master-tidb-0   1/1   Running   <none>/<none>         1     29d   172.16.30.10
aylei-master-tidb-1   1/1   Running   <none>/<none>         37    28d   172.16.30.7
aylei-master-tidb-2   1/1   Running   <none>/<none>         36    28d   172.16.30.9
aylei-master-tidb-3   1/1   Running   <none>/<none>         41    28d   172.16.10.11

VOLUME              CLAIM                      STATUS   CAPACITY   NODE           LOCAL
local-pv-1896e6ce   pd-aylei-master-pd-2       Bound    278Gi      172.16.10.14   /mnt/sas-disks/vol2
local-pv-2edc6512   pd-aylei-master-pd-1       Bound    1861Gi     172.16.30.28   /mnt/sas-disks/vol1
local-pv-482a865d   tikv-aylei-master-tikv-3   Bound    278Gi      172.16.10.5    /mnt/sas-disks/vol1
local-pv-5bbb9350   tikv-aylei-master-tikv-2   Bound    278Gi      172.16.10.12   /mnt/sas-disks/vol2
local-pv-5e34b56e   tikv-aylei-master-tikv-0   Bound    278Gi      172.16.10.19   /mnt/sas-disks/vol3
local-pv-6a1229d6   tikv-aylei-master-tikv-4   Bound    278Gi      172.16.10.11   /mnt/sas-disks/vol3
local-pv-84f137b4   pd-aylei-master-pd-0       Bound    1861Gi     172.16.30.29   /mnt/sas-disks/vol2
local-pv-fda68cc8   tikv-aylei-master-tikv-1   Bound    1861Gi     172.16.30.30   /mnt/sas-disks/vol0
```

### CODE REVIEW

**Comment by zhouqiang-cl** [comment]
please fix ci

**Comment by aylei** [comment]
> please fix ci

This a draft PR and I'm still working on this, hopefully I will fix it before tomorrow

File: cmd/tkctl/main.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd"
+	"github.com/spf13/pflag"
+
```
Comment (weekface) [inline cmd/tkctl/main.go]: remove this empty line.

File: pkg/tkctl/cmd/cmd.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/ctop"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/debug"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/get"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/info"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/list"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/pdctl"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/cmd/use"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/config"
+	"github.com/spf13/cobra"
+	"io"
```
Comment (weekface) [inline pkg/tkctl/cmd/cmd.go]: go fmt

File: pkg/tkctl/cmd/ctop/ctop.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+// limitations under the License.
+
+package ctop
+
+import (
+	"fmt"
+	"github.com/pingcap/tidb-operator/pkg/tkctl/config"
+	"github.com/spf13/cobra"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"os"
```
Comment (weekface) [inline pkg/tkctl/cmd/ctop/ctop.go]: go fmt

File: pkg/tkctl/cmd/list/list.go [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+`
+	listExample = `
+		# list all clusters and sync to local config
+		tkc list
+
+		# filter by namespace
+		tkc list --namespace=foo
+
+		# get tidb cluster in all namespaces
+		tkc list -A
```
Comment (weekface) [inline pkg/tkctl/cmd/list/list.go]: tkc list --all-namespace ?

File: pkg/tkctl/config/config.go [hunk 5]
```diff
@@ -0,0 +1,265 @@
...
+	"github.com/golang/glog"
+	"gopkg.in/yaml.v2"
+	"io/ioutil"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/cli-runtime/pkg/genericclioptions/resource"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"os"
+	"os/user"
+	"path/filepath"
```
Comment (weekface) [inline pkg/tkctl/config/config.go]: go fmt

File: pkg/tkctl/cmd/list/list.go (hunk 4 above)
Comment (aylei) [reply pkg/tkctl/cmd/list/list.go]: `--all-namespaces` is the longer version of `-A`

File: pkg/tkctl/cmd/cmd.go (hunk 2 above)
Comment (aylei) [reply pkg/tkctl/cmd/cmd.go]: I've ran `go fmt` and these seems not being considered as format errors.

**Pull request merged**